src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1359,7 +1361,338 @@ internal static bool IsComObject(object obj)
             return obj != null && Marshal.IsComObject(obj);
 #endif
         }
+
+        #region Implicit Remoting Batching
+
+        // Commands allowed to run on target remote session along with implicit remote commands
+        private static readonly HashSet<string> AllowedCommands = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        {
+            "ForEach-Object",
+            "Measure-Command",
+            "Measure-Object",
+            "Sort-Object",
+            "Where-Object"
```

Comment:
Could you please explain why this commands was included and others (like `Group-Object`, `Select-String`, `Select-Object`, etc.) not?

Reply:
No reason.  This is experimental and will change as needed.